src/lib/components/chat/ChatWindow.svelte

```diff
@@ -45,6 +46,8 @@
 		dispatch("message", message);
 		message = "";
 	};
+
+	$: lastIsError = messages[messages.length - 1]?.from === "user" && !loading;
```

Comment:
The logic check for displaying the error is here, I think that covers all use cases. 

Basically if the last message is from a user but it's not currently loading a new message, it means something went wrong.

Reply:
Ok 👀, looks a bit weird but if it works :)